### PR TITLE
API endpoints for listing and deleting locations

### PR DESCRIPTION
**Pull request by mastoffel**
Added API endpoints for locations:

1) Delete location schema 
2) Delete location by coordinates 
3) List location schemas 
4) List location identifiers

+ tests for all four.

### CODE REVIEW

File: dtbase/backend/api/location/routes.py [hunk 1]
```diff
@@ -148,3 +148,63 @@ def list_locations(schema_name):
...
+def delete_location_schema(schema_name):
+    """
+    Delete a location schema from the database.
+    Endpoint URL: /delete_location_schema/<schema_name>
+    """
+
+    locations.delete_location_schema(schema_name=schema_name, session=db.session)
+    db.session.commit()
+
+    # Check if the schema still exists in the database
```
Comment (mhauru) [inline dtbase/backend/api/location/routes.py:185]: Do we need this test to be run every time the API endpoint gets called? I might have this just as a test in the test suite.

**Review by mhauru — COMMENTED** [review]
This looks great, thanks Martin! I left a couple of minor comments, otherwise good to go.

File: dtbase/backend/api/location/routes.py (hunk 1 above)
Comment (mhauru) [reply dtbase/backend/api/location/routes.py:185]: If the issue is that `locations.delete_location_schema` fails silently if we try to e.g. delete a non-existent schema, then maybe that could be dealt with in the `locations` module without an extra database query somehow?

Comment (mastoffel) [reply dtbase/backend/api/location/routes.py]: Yes, that makes sense. I removed it and now return a boolean from the locations module to check if rows have been deleted.

File: dtbase/backend/api/location/routes.py [hunk 2]
```diff
@@ -148,3 +148,63 @@ def list_locations(schema_name):
...
+    schemas = locations.list_location_schemas(session=db.session)
+    schema_names = [schema["name"] for schema in schemas]
+
+    if schema_name not in schema_names:
+        return jsonify({"status": "success", "message": f"Location schema '{schema_name}' has been deleted."}), 200
+    else:
+        return jsonify({"status": "error", "message": f"Location schema '{schema_name}' not found or could not be deleted."}), 404
+
+
+@blueprint.route("/delete_location_by_coordinates/<schema_name>", methods=["DELETE"])
```
[the remark this answers is not in the document]
Comment (mastoffel) [reply dtbase/backend/api/location/routes.py:193]: I wondered about this! Renamed to delete_location now.

File: dtbase/backend/api/location/routes.py [hunk 3]
```diff
@@ -179,27 +180,41 @@ def delete_location_schema(schema_name):
...
+                {
+                    "status": "error",
+                    "message": f"Location schema '{schema_name}' not found or could not be deleted.",
+                }
+            ),
+            404,
+        )
 
 
 @blueprint.route("/delete_location_by_coordinates/<schema_name>", methods=["DELETE"])
```
Comment (mhauru) [inline dtbase/backend/api/location/routes.py:211]:
```suggestion
@blueprint.route("/delete_location/<schema_name>", methods=["DELETE"])
```

File: dtbase/core/locations.py [hunk 4]
```diff
@@ -286,9 +286,16 @@ def delete_location_schema(schema_name, session=None):
...
     )
-    session.execute(
+
+    # Get the number of rows affected by the delete operation
+    deleted_rows = session.execute(
         sqla.delete(LocationSchema).where(LocationSchema.name == schema_name)
-    )
+    ).rowcount
+
+    print(f"The number of deleted rows is: {deleted_rows}")
```
Comment (mhauru) [inline dtbase/core/locations.py:295]: Was this mainly meant for debugging, or are we planning to leave this?

File: dtbase/tests/test_api_locations.py [hunk 5]
```diff
@@ -213,7 +213,10 @@ def test_delete_location_by_coordinates(client):
         client.post("/location/insert_location/test-schema", json=json.dumps(location))
 
         # Test delete_location
-        response = client.delete("/location/delete_location_by_coordinates/test-schema", json=json.dumps(location))
+        response = client.delete(
+            "/location/delete_location_by_coordinates/test-schema",
```
Comment (mhauru) [inline dtbase/tests/test_api_locations.py:217]:
```suggestion
            "/location/delete_location/test-schema",
```

File: dtbase/core/locations.py (hunk 4 above)
Comment (mhauru) [inline dtbase/core/locations.py:295]:
```suggestion
```

**Pull request merged**